plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,503 @@
+		r.Servers = []string{"tcp://localhost:26379"}
+	}
+
+	r.clients = make([]*RedisSentinelClient, len(r.Servers))
+
+	tlsConfig, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	for i, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("unable to parse to address %q: %v", serv, err)
+		}
+
+		if u.Scheme != "tcp" && u.Scheme != "unix" {
+			return fmt.Errorf("invalid scheme %q. expected tcp or unix", u.Scheme)
+		}
+
+		password := r.Password
+
+		var address string
+		if u.Scheme == "unix" {
+			address = u.Path
+		} else {
+			address = u.Host
+		}
+
+		sentinel := redis.NewSentinelClient(
+			&redis.Options{
+				Addr:      address,
+				Password:  password,
+				Network:   u.Scheme,
+				PoolSize:  1,
+				TLSConfig: tlsConfig,
+			},
+		)
+
+		tags := map[string]string{}
+		if u.Scheme == "unix" {
+			tags["socket"] = u.Path
+		} else {
+			tags["source"] = u.Hostname()
+			tags["port"] = u.Port()
+		}
+
+		r.clients[i] = &RedisSentinelClient{
+			sentinel: sentinel,
+			tags:     tags,
+		}
+	}
+
+	return nil
+}
+
+// Redis list format has string key/values adjacent, so convert to a map for easier use
+func toMap(vals []interface{}) map[string]string {
+	m := make(map[string]string)
+
+	for idx := 0; idx < len(vals)-1; idx += 2 {
+		key, keyOk := vals[idx].(string)
+		value, valueOk := vals[idx+1].(string)
+
+		if keyOk && valueOk {
+			m[key] = value
+		}
+	}
+
+	return m
+}
+
+// Reads stats from all configured servers accumulates stats.
+// Returns one of the errors encountered while gather stats (if any).
+func (r *RedisSentinel) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, client := range r.clients {
+		wg.Add(1)
+
+		go func(acc telegraf.Accumulator, client *RedisSentinelClient) {
+			defer wg.Done()
+
+			gatherMasterStats(acc, client)
+			gatherInfoStats(acc, client)
+		}(acc, client)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+func gatherInfoStats(acc telegraf.Accumulator, client *RedisSentinelClient) {
+	infoCmd := redis.NewStringCmd("info", "all")
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(infoCmd)
+
+	info, infoErr := infoCmd.Result()
+	if infoErr != nil {
+		acc.AddError(infoErr)
+		return
+	}
+
+	rdr := strings.NewReader(info)
+	infoTags, infoFields, err := convertSentinelInfoOutput(client.baseTags(), rdr)
+	if err == nil {
+		acc.AddFields(measurementSentinel, infoFields, infoTags)
+	} else {
+		acc.AddError(err)
+	}
+}
+
+func gatherMasterStats(acc telegraf.Accumulator, client *RedisSentinelClient) {
+	mastersCmd := redis.NewSliceCmd("sentinel", "masters")
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(mastersCmd)
+
+	masters, mastersErr := mastersCmd.Result()
+	if mastersErr != nil {
+		acc.AddError(mastersErr)
+		return
+	}
+
+	for _, master := range masters {
+		master, masterOk := master.([]interface{})
+		if !masterOk {
+			continue
```

Comment:
Should we log this incident?

Reply:
@srebhan Definitely. Added in https://github.com/influxdata/telegraf/pull/10042/commits/f4f7d5e7d32dc47810d63a0d52859b7c58987517.